src/rgw/rgw_admin.cc

```diff
@@ -1,1011 +1,55 @@
 // -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
 // vim: ts=8 sw=2 smarttab
 
-#include <errno.h>
+#include <cerrno>
 #include <iostream>
 #include <sstream>
+#include <cstdlib>
 #include <string>
 
 #include <boost/optional.hpp>
+#include <unordered_set>
+#include <map>
 
 #include "auth/Crypto.h"
 #include "compressor/Compressor.h"
 
-#include "common/armor.h"
 #include "common/ceph_json.h"
 #include "common/config.h"
-#include "common/ceph_argparse.h"
-#include "common/Formatter.h"
-#include "common/errno.h"
-#include "common/safe_io.h"
 
 #include "cls/rgw/cls_rgw_client.h"
 
 #include "global/global_init.h"
 
-#include "include/utime.h"
-#include "include/str_list.h"
 
 #include "rgw_user.h"
 #include "rgw_bucket.h"
-#include "rgw_rados.h"
-#include "rgw_acl.h"
 #include "rgw_acl_s3.h"
 #include "rgw_lc.h"
-#include "rgw_log.h"
-#include "rgw_formats.h"
 #include "rgw_usage.h"
 #include "rgw_replica_log.h"
 #include "rgw_orphan.h"
 #include "rgw_sync.h"
-#include "rgw_sync_log_trim.h"
 #include "rgw_data_sync.h"
 #include "rgw_rest_conn.h"
 #include "rgw_realm_watcher.h"
 #include "rgw_role.h"
 #include "rgw_reshard.h"
 
-
-#define dout_context g_ceph_context
-#define dout_subsys ceph_subsys_rgw
```

Comment:
please don't remove those, they are needed for logging into rgw log file.

Reply:
Those are now defined in rgw_admin_common.h.

Reply:
we prefer to define them in source files to limit their scope